pkg/ccl/storageccl/external_sst_reader.go

```diff
@@ -76,44 +90,73 @@ func ExternalSSTReader(
 				return nil, err
 			}
 		}
-		return storage.NewMemSSTIterator(content, false)
+		inMemorySSTs = append(inMemorySSTs, content)
 	}
+	return storage.NewPebbleMultiMemSSTIterator(inMemorySSTs, false, iterOps)
+}
 
-	raw := &sstReader{
-		ctx:  ctx,
-		sz:   sizeStat(sz),
-		body: f,
-		openAt: func(offset int64) (ioctx.ReadCloserCtx, error) {
-			reader, _, err := e.ReadFileAt(ctx, basename, offset)
-			return reader, err
-		},
+// ExternalSSTReader returns a PebbleSSTIterator for the SSTs in external storage,
+// optionally decrypting with the supplied parameters.
+//
+// ctx is captured and used throughout the life of the returned iterator, until
+// the iterator's Close() method is called.
+func ExternalSSTReader(
+	ctx context.Context,
+	e []cloud.ExternalStorage,
+	filenames []string,
```

Comment:
nit: seems like we'd want to bundle these in a struct.

Reply:
Done.